### PR TITLE
[WFLY-13586] Add a Galleon layer for JSON Binding support

### PR DESCRIPTION
**Pull request by yersan**
Jira issue: https://issues.redhat.com/browse/WFLY-13586

### CODE REVIEW

File: jaxrs/WFLY-13586_Galleon_layer_jsonb.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+* `javax.json.bind.api`: It provisions the JBoss Modules module for JSON-B API.
+* `org.eclipse.yasson`: It provisions the JBoss Modules module for JSON-B implementation.
+
+Currently, there are two subsystems that declare runtime package dependencies that make possible the provision of the JSON-B modules:
+
+* The EE subsystem, which declares JSON-B API and Impl. as optional runtime package dependencies.
+* The JAXRS subsystem, in this case via the registration of `org.jboss.resteasy.resteasy-json-binding-provider` package as an optional runtime package dependency, which depends on JSON-B API and Impl. modules.
+
+We want to make JSON-B an optional layer for JAXRS so users can exclude it if they want. To make it possible, we need to remove the runtime package registration done by the EE subsystem and make EE layer depend optionally on JSON-B. For the JAXRS subsystem counterpart, we will make the `org.jboss.resteasy.resteasy-json-binding-provider` package registration done by JAXRS subsystem passive, so it will be registered only if the JSON-B package dependencies are satisfied by the presence of JSON-B layer. JAXRS layer will depend optionally on JSON-B via EE layer.
```
Comment (marekkopecky) [inline jaxrs/WFLY-13586_Galleon_layer_jsonb.adoc:25]: @yersan (cc @fabiobrz @honza-kasik )AFAIU, there will be two options:

* JSON-B layer will be included, resteasy uses resteasy-json-binding-provider by default for json
* JSON-B layer won't be included, resteasy uses jackson2 for json

Is this intended? Basically, server handles json in http requests in both cases.

Comment (yersan) [reply jaxrs/WFLY-13586_Galleon_layer_jsonb.adoc:25]: Hi @marekkopecky

Yes, server would be able o handle json on http requests. The JSON-B layer brings in the JSR 367(JSON-B) API and Implementation as well as the RestEasy provider for jsonb. If you exclude jsonb layer from your provisioning you won't be able to use the JSR 367 to serialize an object into a Json format. It is unrelated with the fact whether you are able to handle json in http requests.

Maybe I could have misunderstood your concern about it, could you expand it if you see a problem with this exclusion?

Comment (marekkopecky) [reply jaxrs/WFLY-13586_Galleon_layer_jsonb.adoc:25]: > It is unrelated with the fact whether you are able to handle json in http requests.

Ok, so this feature is about JSON-B only.

> Maybe I could have misunderstood your concern about it, could you expand it if you see a problem with this exclusion?

I'm just thinking whether "json" layer wouldn't be better for users than JSON-B layer. I guess that JSON-B layer will be extra small and can do the same as "-Dresteasy.preferJacksonOverJsonB" system property. Anyway, I'm not the one who define requirements.

Comment (yersan) [reply jaxrs/WFLY-13586_Galleon_layer_jsonb.adoc:25]: Thanks @marekkopecky any feedback is very valuable here, so thanks for sharing. The reason of this layer is a try to align with a similar thorntail fraction. That's the reason of this one.

**Comment by darranl** [comment]
For all new layers I think we need to make sure security is considered in the requirements,

We don't want to accidentally leave anything unsecured by default but ideally if layers can avoid a dependency on any security layers that would be preferred.

Failing that for the new layers if security dependencies are required we should be preferring elytron with an option to select legacy security.  At some point in the future all legacy security related layers will be removed.

Can we please add a check for these questions in the requirements?

**Review by fabiobrz — APPROVED** [review]

**Pull request merged**